models.py

```diff
@@ -1,7 +1,152 @@
-import threading
+"""
+Models for Inventory Management Service
+
+All of the models are stored in this module
+
+Models
+------
+ProductInformation - An Inventory entry used in the service
+
+Attributes:
+-----------
+prod_id         (int)       - the id of the product
+prod_name       (string)    - the name of the product
+new_qty         (int)       - quantity of the new product
+used_qty        (int)       - quantity of the used product
+open_boxed_qty  (int)       - quantity of the open-boxed product
+restock_level   (int)       - when the product total quantity reaches below this number,
+                              an automatic restock will be trigger.
+restock_amt     (int)       - the amount of new products restocked
+                              when the total quantity goes under restock_level
+
+"""
+
+# import threading
+from flask_sqlalchemy import SQLAlchemy
+
+# Default ProductInformation property value
+DEFAULT_NEW_QTY = 0
+DEFAULT_USED_QTY = 0
+DEFAULT_OPEN_BOXED_QTY = 0
+DEFAULT_RESTOCK_LEVEL = -1  # -1 means the product won't restock automatically
+DEFALUT_RESTOCK_AMT = 0
+
+PROD_ID = 'prod_id'
+PROD_NAME = 'prod_name'
+NEW_QTY = 'new_qty'
+USED_QTY = 'used_qty'
+OPEN_BOXED_QTY = 'open_boxed_qty'
+RESTOCK_LEVEL = 'restock_level'
+RESTOCK_AMT = 'restock_amt'
 
 class DataValidationError(Exception):
-	pass
+    """ Used for an data validation errors when deserializing """
+    pass
+
+db = SQLAlchemy()
+
+class ProductInformation(db.Model):
```

Comment:
Why changing this from "Entry" to "ProductInformation"? I would personally suggest to simply call it "Inventory". In prof.'s code it is called "Pet". Our API also uses grammar like "GET https://localhost/inventory/123". However nomenclature is not the main concern here. Just though it is a little bit verbose.

Reply:
I think the `Entry` you refers to is the `db` variable in models.py. `ProductInformation` class resembles a "row" in the database.